cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,491 @@ Cask Data Application Platform Release Notes
+
+- :cask-issue:`CDAP-8442` - Fixed an issue in the CDAP UI where the "Stop Program" modal
+  dialog kept loading (showing a spinning wheel) even after the program had been stopped.
+
+- :cask-issue:`CDAP-8445` - The stream endpoint to enqueue messages now returns a 503
+  instead of a 500 if it failed because the dataset service was unavailable.
+
+- :cask-issue:`CDAP-8446` - Fixed an issue where the Transactional.run method could throw
+  the wrong exception if the transaction service was unavailable when it was finishing a
+  transaction.
+
+- :cask-issue:`CDAP-8448` - In general, fixed the HTTP RESTful endpoints to return a 503
+  instead of 500 when the transaction service was unavailable.
+
+- :cask-issue:`CDAP-8509` - Fixed an issue in the Transactional Messaging System (TMS)
+  table upgrade, where the TMS table could be left in a disabled state if the upgrade tool
+  is run after an upgraded CDAP Master is started and then stopped.
+
+- :cask-issue:`CDAP-8544` - Lowered the RPC timeout and number of retries for the HBase
+  operations performed by CDAP Master services.
+
+- :cask-issue:`CDAP-8628` - Fixed an issue in the log saver and the metrics processor that
+  if an exception was thrown during the changing of the number of instances, a container JVM
+  process could be left running without performing any work.
+
+- :cask-issue:`CDAP-8634` - Corrected the Javadoc of the PluginConfig's containsMacro()
+  method to reflect that it always returns false at runtime.
+
+- :cask-issue:`CDAP-8636` - Fixed an issue with Spark programs not working against CDH
+  5.8.4.
+
+- :cask-issue:`CDAP-8672` - Fixed the CDAP Router so that it does not log an error when it
+  cannot discover a service. Previously, the message was logged at the debug level.
+
+- :cask-issue:`CDAP-8687` - Fixed an issue where a user who attempts to create an existing
+  stream that was created by a different user received all the privileges and the original
+  user had their privileges revoked.
+
+- :cask-issue:`CDAP-8694` - Fixed an issue with properly-locating CDAP_HOME in Distributed
+  CDAP instances outside the default ``/opt/cdap`` directory.
+
+- :cask-issue:`CDAP-8753` - Deprecated the ``waitForFinish()`` method in the ProgramManager.
+
+- :cask-issue:`HYDRATOR-1085` - Fixed an issue where the File Sink plugin was failing when
+  writing byte array records.
+
+- :cask-issue:`HYDRATOR-1096` - Fixed an issue with the macro substitution of a Table
+  dataset name.
+
+- :cask-issue:`HYDRATOR-1158` - Fixed an issue with the JSON parser failing if no data was
+  present for a nullable field.
+
+- :cask-issue:`HYDRATOR-1212` - Fixed an issue where runtime arguments were not being
+  passed correctly for the preview run in the CDAP UI.
+
+- :cask-issue:`HYDRATOR-1219` - Fixed an issue in the Wrangler transform with the handling
+  of escaped characters.
+
+- :cask-issue:`HYDRATOR-1226` - Fixed an issue where previews would not run in a
```

Comment:
let's call it Pipeline Preview, since there's also a Preview in the Explore world